server/src/main/java/io/unitycatalog/server/persist/ModelRepository.java

```diff
@@ -0,0 +1,361 @@
+package io.unitycatalog.server.persist;
+
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.model.*;
+import io.unitycatalog.server.persist.dao.CatalogInfoDAO;
+import io.unitycatalog.server.persist.dao.RegisteredModelInfoDAO;
+import io.unitycatalog.server.persist.dao.SchemaInfoDAO;
+import io.unitycatalog.server.persist.utils.FileUtils;
+import io.unitycatalog.server.persist.utils.HibernateUtils;
+import io.unitycatalog.server.persist.utils.PagedListingHelper;
+import io.unitycatalog.server.persist.utils.RepositoryUtils;
+import io.unitycatalog.server.utils.ValidationUtils;
+import java.util.*;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ModelRepository {
+  private static final ModelRepository INSTANCE = new ModelRepository();
+  private static final Logger LOGGER = LoggerFactory.getLogger(ModelRepository.class);
+  private static final SessionFactory SESSION_FACTORY = HibernateUtils.getSessionFactory();
+  private static final SchemaRepository SCHEMA_REPOSITORY = SchemaRepository.getInstance();
+  private static final PagedListingHelper<RegisteredModelInfoDAO> LISTING_HELPER =
+      new PagedListingHelper<>(RegisteredModelInfoDAO.class);
+
+  private ModelRepository() {}
+
+  public static ModelRepository getInstance() {
+    return INSTANCE;
+  }
+
+  public RegisteredModelInfo getRegisteredModelById(String registeredModelId) {
+    LOGGER.info("Getting registered model by id: " + registeredModelId);
+    try (Session session = SESSION_FACTORY.openSession()) {
+      session.setDefaultReadOnly(true);
+      Transaction tx = session.beginTransaction();
+      try {
+        RegisteredModelInfoDAO registeredModelInfoDAO =
+            session.get(RegisteredModelInfoDAO.class, UUID.fromString(registeredModelId));
+        if (registeredModelInfoDAO == null) {
+          throw new BaseException(
+              ErrorCode.NOT_FOUND, "Registered model not found: " + registeredModelId);
+        }
+        RegisteredModelInfo registeredModelInfo = registeredModelInfoDAO.toRegisteredModelInfo();
+        SchemaInfoDAO schemaInfoDAO =
```

Comment:
Do we really need to check for these, unless we did some dangling/ghost deletes where we deleted parent schema/catalog without deleting the child entities ?

Reply:
I guess we don't need the check if we're positive that situation could never occur, but if this does happen we would be throwing a less specific exception/message.

Reply:
Sure, we can keep it. But we are anyways handling deleting children entities properly in delete.